### ISSUE TITLE
How do I use radio buttons properly?

### ISSUE THREAD

**Issue by tsquez** [wpmetabox/meta-box#1199]
I posted this question in the support forum over at MetaBox.io under "[Using Radio Fields Properly](https://metabox.io/support/topic/using-radio-fields-properly/)". 

Here is what i wrote (if you cannot get into the forums):

> Quick question. I cannot seem to wrap my head around using radio buttons and displaying the information on the frontend. Checkboxes are OK, I can do those all day long with no problem. But for this particular metabox, I need to use radio buttons for the options. Here is what I have:

	//Featured Image Options
	function tp_primo_featured_image_options_custom_meta_box( $meta_boxes ) {
		$prefix = 'tp_primo_';
		$meta_boxes[] = array(
			'id' => 'tp_primo-featured-image-options',
			'title' => esc_html__('Featured Image Options','tp_primo'),
			'post_types' => array('post','page'),
			'context' => 'side',
			'priority' => 'default',
			'autosave' => true,
			'fields' => array(
				array(
					'id' => $prefix . 'featured_image_layout_options',
					'type' => 'radio',
					'placeholder' => '',
					'options' => array(
						'show_below_title' => esc_html__('Featured Image Below Title','tp_primo'),
						'show_above_title' => esc_html__('Featured Image Above Title','tp_primo'),
						'page_header_contained' => esc_html__('Page Header Contained.','tp_primo'),
						'page_header_contained_title_overlay' => esc_html__('Page Header Cont. w/Title','tp_primo'),
						'page_header_contained_title_overlay_parallax' => esc_html__('Page Header Cont. w/Title - Parallax','tp_primo'),
						'page_header_full_width' => esc_html__('Page Header Full WIdth','tp_primo'),
						'page_header_full_width_title_overlay' => esc_html__('Page Header FW w/Title','tp_primo'),
						'page_header_full_width_title_overlay_parallax' => esc_html__('Page Header FW w/Title - Parallax','tp_primo')
					),
					'inline' => false,
					'std' => 'show_blow_title',
				),
				array(
					'id' => $prefix . 'page_header_height',
					'type' => 'text',
					'name' => esc_html__('Page Header Height','tp_primo'),
					'desc' => esc_html__('Use PX, %, REM, EM and VH based values.','tp_primo'),
				),
			),
		);
		return $meta_boxes;
	}
	add_filter('rwmb_meta_boxes','tp_primo_featured_image_options_custom_meta_box');

> OK so the Featured Image Options metabox shows up fine. As you can see, the default is set to `show_below_title`. This means that the featured image will display below the title of the post/page. Should be easy enough right? But nooooooo…lol!
> 
> So if the default is `show_below_title`, how do I get the featured image to show under the post/page title? This is what I would like to use in the theme if possible (if not oh well):

	<?php if (what_is_the_code_that_goes_here)): ?>
		<?php do_action('totalpress_featured_image_single'); ?>
	<?php endif; ?>

> would that even be correct?
> 
> Now before anyone starts posting read this, or read that, I did read the following:
> 
> https://docs.metabox.io/fields/radio/
> 
> https://docs.metabox.io/rwmb-meta/
> 
> https://docs.metabox.io/rwmb-the-value/
> 
> and I did do an extensive search of the forums and for the life of me I cannot find an answer and therefore cannot figure out what to do.
> 
> I do apologize if the answer is “stupid” simple and I am missing it, I am really new to metaboxes but I am finding that they are freaking awesome to use.
> 
> Any help is greatly appreciated.
> 
> Hope everyone has a Merry Christmas

Now I got an answer from Truong Giang, who is a moderator. This is what he posted:

> Hi there,
> Please try this code:

	$option = rwmb_meta( 'tp-primo-featured-image-options' );
	if ( ! $option || 'show_below_title' == $option ) {
	    // do action.
	}

> For other options, you don’t need ! $option
> Thank you.

So I tried using what he posted but I was unable to get it to work. This is what I used:

	if ( ! function_exists('totalpress_build_content_page_entry_header')) :
		function totalpress_build_content_page_entry_header() { ?>

			<?php $option = rwmb_meta('tp-primo-featured-image-options');
			if ( $option || 'show_above_title' == $option ) {
			    do_action('totalpress_featured_image_single');
			} ?>

			<header class="entry-header">	
				<?php if (get_post_meta($post->ID,'totalpress_page_options_hide_title',true)) {
			   		the_title('<h1 class="entry-title hide" itemprop="headline">','</h1>');
				  } else {
			   		the_title('<h1 class="entry-title" itemprop="headline">','</h1>');
				  } ?>
			</header>

			<?php $option = rwmb_meta('tp-primo-featured-image-options');
			if ( ! $option || 'show_below_title' == $option ) {
			    do_action('totalpress_featured_image_single');
			} ?>
			
		<?php
		}
		add_action('totalpress_content_page_entry_header','totalpress_build_content_page_entry_header');
	endif;

Of course I know this is wrong because its not working...lol. Well it does work to get the featured image to display under the title, which is what I asked.

However, I am at a loss for the other options.

Can someone help me figure this out please? I am not really understanding how I am supposed to use the information from the radio fields.  You know like this:

>if 'show_below_title' show this
>if 'show_above_title' show this
>if 'page_header_contained' show this
>if 'page_header_contained_title_overlay' show this
>if 'page_header_contained_title_overlay_parallax' show this
>if 'page_header_full_width' show this
>if 'page_header_full_width_title_overlay' show this
>if 'page_header_full_width_title_overlay_parallax' show this

With check boxes, yes those are easy but these radio fields are very confusing.

Any help getting this to work and help me understand how they actually work is greatly appreciated.

**Comment by tsquez** [comment]
OK so I think I was asking the wrong question here so I have decided to re-ask the question. OK so I created this simple meta box:

	function how_do_I_use_radio_buttons_meta_box( $meta_boxes ) {
			$prefix = 'how_to_use_';

			$meta_boxes[] = array(
				'id' => 'how-to-use-radio-buttons',
				'title' => esc_html__('How To Use Radio Fields','text-domain'),
				'post_types' => array('post','page'),
				'context' => 'side',
				'priority' => 'default',
				'autosave' => true,
				'fields' => array(
					array(
						'id' => $prefix . 'radio_buttons',
						'type' => 'radio',
						'placeholder' => 'boogie',
						'options' => array(
							'button_one_selected' => 'Radio Button One',
							'button_two_selected' => 'Radio Button Two',
							'button_three_selected' => 'Radio Button Three',
							'button_four_selected' => 'Radio Button Four',
							'button_five_selected' => 'Radio Button Five',
						),
						'inline' => false,
					),
				),
			);

			return $meta_boxes;
		}
		add_filter( 'rwmb_meta_boxes', 'how_do_I_use_radio_buttons_meta_box' );

As you can see there are five (5) choices. I have five (5) questions regarding this metabox:

**_First question_**: When a user selects Radio Button One, how would I display “I Clicked on Radio Button One” somewhere in the theme?

**_Second question_**: When a user selects Radio Button Two, how would I display “I Clicked on Radio Button Two” somewhere in the theme?

**_Third question_**: When a user selects Radio Button Three, how would I display the words “I Clicked on Radio Button Three” somewhere in the theme?

**_Fourth question_**: When a user selects Radio Button Four, how would I display the words “I Clicked on Radio Button Four” somewhere in the theme?

**_Fifth question_**: When a user selects Radio Button Five, how would I display the words “I Clicked on Radio Button Five” somewhere in the theme?

So if you could answer those questions, then maybe I can understand how to use radio fields in meta boxes.

**Comment by rilwis** [comment]
Hi, I answered in the premium support forum, but posting here for anyone who has the same question:

Please try to use the following code for your 5 questions:

```php
$value = rwmb_meta( 'how_to_use_radio_buttons' );
if ( 'button_one_selected' == $value ) {
    echo 'I Clicked on Radio Button One';
} elseif ( 'button_two_selected' == $value ) {
    echo 'I Clicked on Radio Button Two';
} elseif ( 'button_three_selected' == $value ) {
    echo 'I Clicked on Radio Button Three';
} elseif ( 'button_four_selected' == $value ) {
    echo 'I Clicked on Radio Button Four';
} elseif ( 'button_five_selected' == $value ) {
    echo 'I Clicked on Radio Button Five';
}
```

Basically, it’s the if..else statement. You just need to check the value of the custom field and output the corresponding text.

**Comment by tsquez** [comment]
I would like to say thank you very much for taking the time and energy to show me how to use a meta box with radio fields. I did however figure it out but it was not with your solution.

Someone on another site who had a tutorial on meta boxes explained what I needed to do precisely and it worked beautifully.

Thanks for all the effort Anh I really do appreciate it.

If anyone would like to know what I did I would be more than happy to let them know.
